File: 프로그래머스/unrated/181851. 전국 대회 선발 고사/전국 대회 선발 고사.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <tuple>
using namespace std;

bool CmpRank(tuple<int, int, bool> v1, tuple<int, int, bool> v2)
{
    return get<1>(v1) <get<1>(v2);
}

int solution(vector<int> rank, vector<bool> attendance) {
    vector<tuple<int, int, bool>> m;
    
    for (int i = 0; i < rank.size(); i++)
        m.push_back(make_tuple(i, rank[i], attendance[i]));
    
    sort(m.begin(), m.end(), CmpRank);
    
    vector<int> ans;
    for (int i = 0; i < m.size(); i++)
        if (get<2>(m[i]))
            ans.push_back(get<0>(m[i]));
    
    return 10000 * ans[0] + 100 * ans[1] + ans[2];
}

File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.java
import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

class Solution {

    public int solution(int[] rank, boolean[] attendance) {
        int answer = 0;

        List<Pairs> list = new ArrayList<>();
        for (int i = 0; i < rank.length; i++) {
            list.add(new Pairs(rank[i], attendance[i], i));
        }

        list.sort(Comparator.comparing(Pairs::getRank));

        int[] ints = new int[3];
        int idx = 0;
        for (Pairs p : list) {
            if (p.getAvail()) {
                ints[idx++] = p.getIndex();
                if (idx == 3) break;
            }
        }

        return 10000 * ints[0] + 100 * ints[1] + ints[2];
    }

    static class Pairs {

        Integer rank;
        Boolean avail;
        Integer index;

        public Pairs(Integer rank, Boolean avail, int index) {
            this.rank = rank;
            this.avail = avail;
            this.index = index;
        }


        public Integer getRank() {
            return rank;
        }

        public Boolean getAvail() {
            return avail;
        }

        public Integer getIndex() {
            return index;
        }
    }
}